engine/launcher/src/main/scala/org/enso/launcher/internal/OS.scala

```diff
@@ -0,0 +1,35 @@
+package org.enso.launcher.internal
+
+sealed trait OS
+object OS {
+  case object Linux   extends OS
+  case object MacOS   extends OS
+  case object Windows extends OS
+
+  /**
+    * Checks if the application is being run on Windows.
+    */
+  def isWindows: Boolean =
+    operatingSystem == OS.Windows
+
+  /**
+    * Returns which [[OS]] this program is running on.
+    */
+  def operatingSystem: OS = {
+    val name = System.getProperty("os.name").toLowerCase
+    if (name.contains("linux")) OS.Linux
+    else if (name.contains("mac")) OS.MacOS
+    else if (name.contains("windows")) OS.Windows
+    else
+      throw new RuntimeException(
+        s"fatal error: os.name `$name` is not recognized."
+      )
+  }
```

Comment:
This logic seems quite brittle to me. Is this the canonical way of detecting OS? It's also not extensible easily to other systems.

Reply:
I'm not sure if a 'canonical' way is even defined but that seems to be more or less how it is done across different projects, for example: [Apache Commons](http://commons.apache.org/proper/commons-lang/javadocs/api-release/src-html/org/apache/commons/lang3/SystemUtils.html#line.1119) or [this StackOverflow answer](https://stackoverflow.com/questions/228477/how-do-i-programmatically-determine-operating-system-in-java) (although the Linux check is a bit more sophisticated there, maybe it is worth adapting).

What exactly do you mean by extensibility here? If we want to support another platform, we can just add it here. And we need to do such an addition manually, because we do have system-specific logic that will need updating (the logic determining installation paths).

Or maybe we should provide a fallback for unknown OS that assumes it to be some kind of Unix flavour? I think this is the best we can do without further knowledge about an unknown OS.